### PR TITLE
fix: Convert to ESM

### PR DESCRIPTION
**Pull request by benmccann**
**Marked version:** `master`

**Markdown flavor:** all

## Description

Converts the source code to ESM to be more modern and allow Rollup to better handle it. Rollup will still convert the final code to be UMD / ESM, but will do so more efficiently resulting in smaller output files

## Contributor

- [ ] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [X] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

**Comment by benmccann** [comment]
Ok, I've added `"type": "module"` and converted all the tests to ESM as well

> We could probably remove lib/marked.esm.js with this update. Or make it just a file that exports src/marked.js

We could, but I actually think it might be preferable to keep as is so that we know the UMD and ESM versions have the same exposed APIs. I think it could be easier to document and explain that way.

**Comment by benmccann** [comment]
Ok, thanks for letting me know. I didn't have permission to see the Vercel link, so didn't know what was wrong. I've updated it so that it's working now

File: man/marked.1.txt
Comment (calculuschild) [inline man/marked.1.txt:13]: Random extra whitespace added?

File: man/marked.1.txt
Comment (calculuschild) [inline man/marked.1.txt:30]: More random whitespace. I think there's 5 or 6 other places in this PR so double check.

**Review by calculuschild — APPROVED** [review]
Seems fine except for a bunch of random whitespace being be added between words in the documentation.

I really don't have experience in this side of JavaScript with module packaging and stuff though, so I don't know how useful my review is.

File: man/marked.1.txt
Comment (benmccann) [reply man/marked.1.txt]: fixed

File: man/marked.1.txt
Comment (benmccann) [reply man/marked.1.txt]: fixed

**Comment by benmccann** [comment]
I think that might have been from @UziTech 's changes. I'm not sure if perhaps that file is autogenerated from something, but I just manually updated it

**Comment by UziTech** [comment]
That is auto generated but we can edit it manually.

https://github.com/markedjs/marked/blob/e93f800ad610a42897351ed61ab521ab61874a15/Makefile#L12-L13

**Comment by benmccann** [comment]
Cool. Do we need to wait for @davisjam, @joshbruce and @styfle to weigh in before merging?

**Comment by UziTech** [comment]
I was just giving a little bit of time for anyone to weigh in. I'll merge this later today if there is no issues.

**Comment by jimmywarting** [comment]
Ty 4 this!

**Pull request merged**
